src/x11_in_gtk2.c

```diff
@@ -204,8 +205,24 @@ forward_size_request(SuilX11Wrapper* socket,
 		if (hints.flags & PMinSize) {
 			width  = MAX(width, hints.min_width);
 			height = MAX(height, hints.min_height);
+			socket->hints.min_width = hints.min_width;
+			socket->hints.min_height = hints.min_height;
+			gtk_widget_set_size_request(GTK_WIDGET(&socket->socket),
```

Comment:
`forward_size_request` is called from the gtk size-allocate callback.  You must not change the widget's own size-request here.

Reply:
You must reset the size-request after size-allocate as this setting shouldn't resize the widget, it should only allow the user to shrink it to the min size.